src/test/regress/expected/function_propagation.out

```diff
@@ -0,0 +1,331 @@
+(2 rows)
+
+SELECT * FROM run_command_on_workers($$SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_1'::regproc::oid;$$) ORDER BY 1,2;
+ nodename  | nodeport | success |                                               result
+---------------------------------------------------------------------
+ localhost |    57637 | t       | (function,"{function_propagation_schema,func_1}",{function_propagation_schema.function_prop_type})
+ localhost |    57638 | t       | (function,"{function_propagation_schema,func_1}",{function_propagation_schema.function_prop_type})
+(2 rows)
+
+BEGIN;
+    CREATE TYPE function_prop_type_2 AS (a int, b int);
+COMMIT;
+CREATE OR REPLACE FUNCTION func_2(param_1 int)
+RETURNS function_prop_type_2
+LANGUAGE plpgsql AS
+$$
+BEGIN
+    return 1;
+END;
+$$;
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.function_prop_type_2'::regtype::oid;
+                pg_identify_object_as_address
+---------------------------------------------------------------------
+ (type,{function_propagation_schema.function_prop_type_2},{})
+(1 row)
+
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_2'::regproc::oid;
+                pg_identify_object_as_address
+---------------------------------------------------------------------
+ (function,"{function_propagation_schema,func_2}",{integer})
+(1 row)
+
+SELECT * FROM run_command_on_workers($$SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.function_prop_type_2'::regtype::oid;$$) ORDER BY 1,2;
+ nodename  | nodeport | success |                            result
+---------------------------------------------------------------------
+ localhost |    57637 | t       | (type,{function_propagation_schema.function_prop_type_2},{})
+ localhost |    57638 | t       | (type,{function_propagation_schema.function_prop_type_2},{})
+(2 rows)
+
+SELECT * FROM run_command_on_workers($$SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_2'::regproc::oid;$$) ORDER BY 1,2;
+ nodename  | nodeport | success |                           result
+---------------------------------------------------------------------
+ localhost |    57637 | t       | (function,"{function_propagation_schema,func_2}",{integer})
+ localhost |    57638 | t       | (function,"{function_propagation_schema,func_2}",{integer})
+(2 rows)
+
+BEGIN;
+    CREATE TYPE function_prop_type_3 AS (a int, b int);
+COMMIT;
+-- Objects in the body part is not found as dependency
+CREATE OR REPLACE FUNCTION func_3(param_1 int)
+RETURNS int
+LANGUAGE plpgsql AS
+$$
+DECLARE
```

Comment:
what happens if the object is actually used in the function body, and the function is a SQL function?

In general, might be useful to to have a test that ensures we send with `SET LOCAL check_function_bodies TO off`?

Reply:
(if we already have such a test, a link would be useful for future reference as well, thanks)

Reply:
I've added a test for it.